jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/DownloadObjectAction.kt

```diff
@@ -34,39 +64,143 @@ class DownloadObjectAction(private val project: Project, treeTable: S3TreeTable)
         }
     }
 
-    override fun enabled(nodes: List<S3TreeNode>): Boolean = nodes.all { it is S3TreeObjectNode }
+    private fun downloadSingle(project: Project, file: S3TreeObjectNode) {
+        val selectedLocation = getDownloadLocation(foldersOnly = false) ?: return
+
+        val destinationFile = if (selectedLocation.isDirectory()) {
+            selectedLocation.resolve(file.name)
+        } else {
+            selectedLocation
+        }
+
+        val downloads = listOf(DownloadInfo(file.key, destinationFile))
+
+        // If user has requested a single file as their destination, presume they want to overwrite it
```

Comment:
if we presume they want to overwrite it, why are we checking for conflicts?

Reply:
Moved comment down few lines. It applies to the else block where do don't check for conflicts